test/mimc.js

```diff
@@ -0,0 +1,104 @@
+const mimcjs = require("../circomlib/src/mimc7.js");
+const mimcGenContract = require("../circomlib/src/mimc_gencontract.js");
+
+/*
+    Here we want to test the hashing of the MIMC7 contract
+*/ 
+
+contract("MiMC7 Hashing", async accounts => {
+    let mimc;
+    const SEED = "mimc";
+
+    it("Should deploy the contract", async () => {
+        const C = new web3.eth.Contract(mimcGenContract.abi);
+
+        mimc = await C.deploy({
+            data: mimcGenContract.createCode(SEED, 91)
+        }).send({
+            gas: 1500000,
+            from: accounts[0]
+        });
+    });
+
+    it("Should return a hash", async () => {
+        const input = [2];
+        const key = 22;
+
+        let result = await mimc.methods.MiMCpe7(input, key).call();
+        // console.log(result);
+        // console.log(result.toString());
+    });
+
+    it("Should hash value correctly in js", async () => {
+        const m = BigInt(3703141493535563179657531719960160174296085208671919316200479060314459804651).toString();
+        const k = BigInt(918403109389145570117360101535982733651217667914747213867238065296420114726).toString();
+        
+        const jsExpected = mimcjs.hash(m, k);
+        const solidityResult = await mimc.methods.MiMCpe7(m, k).call();
+
+        assert.equal(solidityResult.toString(), jsExpected.toString(), "Unexpected result");
+    });
+
+    it("Should hash value correctly in solidity", async () => {
+        const res = await mimc.methods.MiMCpe7(1,2).call();
+        const res2 = await mimcjs.hash(1,2,91);
+
+        assert.equal(res.toString(), res2.toString());
+    });
+
+    
+    /*
+
+              G
+            E    F
+          A  B  C  D
+
+        E = hash(A, B)
+        F = hash(C, D)
+        G = hash(E, F)
+
+        multihash(a,b,c,d) != G
+        multihash(a,b,c,d) != hash(hash(a, b), hash(c, d))
+        multihash(a,b,c,d) == hash(hash(hash(hash(IV, a), b), c), d)
```

Comment:
```
              G          <-- merkle root
            E    F
          A  B  C  D     <-- merkle tree leaves
```
Consider the above Merkle tree, where `E = hash(A, B)`, `F = hash(C, D)`, and `G = hash(E, F)`.

We could have a function called `multiHashTree`, such that `multiHashTree(A, B, C, D) == G`, noting that `G` is the hash of the Merkle tree root. Is this something we might need?

Reply:
Could be useful, and we can easily implement it once the need comes!